src/main/features/directions-questionnaire/routes/hearing-location.ts

```diff
@@ -97,17 +114,38 @@ export default express.Router()
             const court: Court = await getNearestCourt(form.model.alternativePostcode)
             if (court !== undefined) {
               const alternativeCourtDetails: CourtDetails = await getCourtDetails(court.slug)
+              draft.document.hearingLocation.alternativeCourtName = court.name
+              draft.document.hearingLocation.alternativePostcode = form.model.alternativePostcode
+              draft.document.hearingLocation.alternativeOption = form.model.alternativeOption
+              draft.document.hearingLocation.facilities = alternativeCourtDetails.facilities
+              draft.document.hearingLocationSlug = court.slug
               form.model = new HearingLocation(court.name, form.model.alternativePostcode, alternativeCourtDetails.facilities)
             }
+            await new DraftService().save(draft, user.bearerToken)
             renderPage(res, form, court === undefined)
+          } else if (form.model.courtAccepted === YesNoOption.NO && form.model.alternativeOption === AlternativeCourtOption.BY_NAME) {
+            draft.document.hearingLocation = form.model
+            draft.document.hearingLocation.alternativeOption = form.model.alternativeOption
+            draft.document.hearingLocation.courtName = form.model.alternativeCourtName
+            draft.document.hearingLocationSlug = ''
 
+            await new DraftService().save(draft, user.bearerToken)
+            res.redirect(Paths.expertPage.evaluateUri({ externalId: res.locals.claim.externalId }))
           } else {
-
-            if (form.model.courtAccepted === undefined) {
-              draft.document.hearingLocation.alternativeCourtName = form.model.alternativeCourtName
-            } else {
-              draft.document.hearingLocation.courtName = form.model.courtName
+            if (form.model.alternativeOption !== undefined && form.model.alternativeOption === AlternativeCourtOption.BY_NAME) {
+              draft.document.hearingLocation.alternativeCourtName = ''
```

Comment:
can all these empty strings be replaced with undefined please?

Reply:
As we discussed before that courtNames are defined as Mandatory .... I have added todo for this to remove or undefined once backend is fixed